benefits/enrollment_switchio/views.py

```diff
@@ -26,9 +30,43 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class GatewayUrlView(EligibleSessionRequiredMixin, View):
+class GatewayUrlView(AgencySessionRequiredMixin, EligibleSessionRequiredMixin, View):
     """View for the tokenization gateway registration"""
 
-    def get(self, request, *args, **kwargs):
-        data = {"gateway_url": "https://server/gateway/uuid"}
+    def get(self, request: HttpRequest, *args, **kwargs):
+        agency = session.agency(request)
+        switchio_config = agency.switchio_config
+
+        client = Client(
+            api_url=switchio_config.api_base_url,
+            api_key=switchio_config.api_key,
+            api_secret=switchio_config.api_secret,
+            private_key=switchio_config.private_key,
+            client_certificate=switchio_config.client_certificate,
+            ca_certificate=switchio_config.ca_certificate,
+        )
+
+        route = reverse(routes.ENROLLMENT_SWITCHIO_INDEX)
+        redirect_url = _generate_redirect_uri(request, route)
+
+        registration = client.request_registration(
+            eshopRedirectUrl=redirect_url,
+            mode="register",
+            eshopResponseMode=EshopResponseMode.FORM_POST,
+            timeout=settings.REQUESTS_TIMEOUT,
+        )
```

Comment:
Need to wrap this in a `try-except` and populate JSON with `redirect` if an exception occurred